### PR TITLE
Switch to yarn

### PR DESCRIPTION
**Pull request by JPrevost**
- documents change of why we are switching to yarn
- switches us to yar
- updates docs on how to run common commads in readme

#### How can a reviewer manually see the effects of these changes?

- `yarn install` on this branch then confirm the app works as expected with `yarn test:unit` and `yarn serve`

#### What are the relevant tickets?
- https://mitlibraries.atlassian.net/browse/DISCO-78

#### Includes new or updated dependencies?
NO [not directly, but externally it now expects yarn instead of npm]

### CODE REVIEW

**Review by jazairi — APPROVED** [review]
Dependencies install, tests pass, and searches execute. :shipit: I'm noticing some curious Privacy Badger behavior re: TIMDEX that I'll share in Slack.

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,7 @@
 .DS_Store
 node_modules
 /dist
+coverage
```
Comment (matt-bernhardt) [inline .gitignore:4]: Is coverage part of this work?

Comment (JPrevost) [reply .gitignore:4]: Shhhh (no, but that was easier to do now due to my local status)

**Review by matt-bernhardt — APPROVED** [review]
Question about coverage aside, I"m +1 to this. It builds locally, tests pass, and the app runs. I'm hopeful that this will resolve some of the issues we've been running into while using NPM.

:shipit:

**Pull request merged**